torch/csrc/autograd/anomaly_mode.cpp

```diff
@@ -0,0 +1,50 @@
+#include "torch/csrc/autograd/anomaly_mode.h"
+#include "torch/csrc/utils/auto_gil.h"
+#include "torch/csrc/utils/python_strings.h"
+
+#include "stdexcept"
+#include "iostream"
+
+namespace torch { namespace autograd {
+
+bool AnomalyMode::_enabled = 0;
+PyObject* AnomalyMode::stacktrace_key = THPUtils_packString("stacktrace_");
+
+void AnomalyMode::store_stack(PyObject* metadata) {
+  AutoGIL gil;
+  auto mod = PyImport_ImportModule("traceback");
+  if (!mod) {
+    return;
```

Comment:
python_error(); ?

Reply:
What would that do?
I admit I'm not sure how this should be handled.
Like that?
```
if (!mod) {
  throw python_error();
}
```

Reply:
that's correct.  It will give you a nice python backtrace and error message.